### PR TITLE
feat: allow email domain projects roles

### PR DESCRIPTION
**Pull request by ZeRego**
Closes: #5932 

### Description:

Miro sketch: https://miro.com/app/board/uXjVM-WvQ-s=/?share_link_id=488803571686

- [x] **UI:** allow admin to select the role for each project
- [x] **BE:** migrate column to a table. Ignore project uuids that no longer exists
- [x] **Docs:** add details about project roles. refactor docs to reuse snippet
- [x] **Analytics events:** update the `lightdash_server_organization_allowed_email_domains_updated` to have a list of project roles

Captura de ecrã 2023-06-28, às 16 34 38
Captura de ecrã 2023-06-28, às 16 34 13
Captura de ecrã 2023-06-28, às 16 34 03

https://github.com/lightdash/lightdash/assets/9117144/553a1a80-f8d2-433d-a112-887bfe78554e

### CODE REVIEW

File: packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts [hunk 1]
```diff
@@ -0,0 +1,112 @@
+import { Knex } from 'knex';
+
+const OrganizationAllowedEmailDomainProjectsTableName =
+    'organization_allowed_email_domain_projects';
+const OrganizationAllowedEmailDomainsTableName =
+    'organization_allowed_email_domains';
```
Comment (rephus) [inline packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts:6]: can't we add a new column (project_uuid) on the existing table ?

File: packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+export async function down(knex: Knex): Promise<void> {
+    // add old column
+    await knex.schema.alterTable(
+        OrganizationAllowedEmailDomainsTableName,
+        (tableBuilder) => {
+            tableBuilder
+                .specificType('project_uuids', 'TEXT[]')
+                .notNullable()
+                .defaultTo(JSON.stringify({}));
```
Comment (rephus) [inline packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts:91]: is `TEXT[]'` a valid type for `JSON.stringify({})`  ? shouldn't be '[]' ?  {} sounds more like a jsonblob

**Review by rephus — COMMENTED** [review]
an e2e test would be :kissing: :ok_hand:

**Comment by rephus** [comment]
I've tested this on render and everything works as expected. 

image

I rolled back on render to ensure `down` is working fine, and everything looks good.

**Review by rephus — APPROVED** [review]
Works as expected. I'm only concerned about the migration "type", but if you are happy about it, then we can merge !

File: packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts (hunk 2 above)
Comment (ZeRego) [reply packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts:91]: If you access directly to the db you can see that the value of `TEXT[]` is `{1234-uuid, another-uuid}`. 
iirc I tried using`[]` and it errored

File: packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts (hunk 1 above)
Comment (ZeRego) [reply packages/backend/src/database/migrations/20230628153935_allow-email-domains-projects-table.ts:6]: there are multiple projects per allowed email domains, that's why we needed the separate table.

**Pull request merged**